src/runtime/runtime_rp2350.go

```diff
@@ -2,8 +2,40 @@
 
 package runtime
 
+// #include <stdint.h>
+//
+// static void spinlock_lock(void *lock) {
```

Comment:
What was wrong with the `sync/atomic` operations? Custom assembly for just one target (rp2350) seems to me like a maintenance headache for little gain.

Reply:
I forget, but I'll take a second look at them. One reason was the size of them, but I'm going to take a second look